Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -462,8 +543,15 @@ private void CheckForNewBlock()
         public readonly struct Reader
         {
             [NativeDisableUnsafePtrRestriction] private readonly BufferInfo* m_BufferInfo;
-            [NativeDisableUnsafePtrRestriction] private readonly LaneInfo* m_Lanes;
-
+            
+            /// <summary>
+            /// Reports if this <see cref="Reader"/> has been created and is valid to use.
+            /// </summary>
+            public bool IsCreated
+            {
+                get => m_BufferInfo != null;
```

Comment:
same comment about adding an assert to the constructor.

Reply:
Ensured that constructor can't be called without passing through the assert.